### PR TITLE
Update seccomp e2e test for GA

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

We now execute the seccomp tests with `[LinuxOnly]` as described in the appropriate KEP test plan:
    
https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#test-plan
    
The tests have been changed to used the new seccomp field within the security context in replacement for the annotations.
 
**Which issue(s) this PR fixes**:

Requires #91381 and #91408

/cc @pjbgf @hasheddan @evrardjp
 
**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:
```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Review by pjbgf — COMMENTED** [review]
LGTM, once #91381 is merged, this PR should be good after the seccomp type names are updated.

File: test/e2e/node/security_context.go
Comment (pjbgf) [inline test/e2e/node/security_context.go:195]: @saschagrunert What's your thoughts about checking whether the pod level seccomp applies to the sandbox container?

Comment (saschagrunert) [reply test/e2e/node/security_context.go:195]: Hm, how would we verify that if not from a running container? (like https://github.com/kubernetes/kubernetes/pull/91442/commits/db5843d1b3a23299e3f5fa5b4d121db57d479a3c#diff-d396c518f484bf943b4f92eec90c708bR175)

I think we could make the tests more valuable when bringing real localhost profiles into the game. We could change the test bootstrap process to place the profiles in the nodes. But now I'm wondering which test we would change for that. Probably `pull-kubernetes-e2e-gce` and we could scope the tests with something like `[Seccomp Localhost]`.

Comment (pjbgf) [reply test/e2e/node/security_context.go:195]: > Hm, how would we verify that if not from a running container? (like [db5843d#diff-d396c518f484bf943b4f92eec90c708bR175](https://github.com/kubernetes/kubernetes/commit/db5843d1b3a23299e3f5fa5b4d121db57d479a3c#diff-d396c518f484bf943b4f92eec90c708bR175))

To test we could use a localhost profile at pod level that would block the sandbox requirements - or simply a default action of ERRNO. Last time I mapped sandbox's profile required:
https://github.com/pjbgf/kubernetes/blob/a3bf173f20904a8ee70bbc9526dd5b605c511757/cmd/kubelet/app/options/container_runtime.go#L46

Then ensure that the pod would run if the sandbox was not blocked, by overwritting the seccomp of the other container running in that pod by setting its SeccompProfile to Unconfined or RuntimeDefault. If pod fails to run, the seccomp was applied to the sandbox.

> I think we could make the tests more valuable when bringing real localhost profiles into the game. We could change the test bootstrap process to place the profiles in the nodes. But now I'm wondering which test we would change for that. Probably `pull-kubernetes-e2e-gce` and we could scope the tests with something like `[Seccomp Localhost]`.

Sounds good.

Comment (saschagrunert) [reply test/e2e/node/security_context.go:195]: I think we have the "overwrite pod sandbox profile in container" scenario already covered, but with localhost profiles it would be better to test. I'll check out what is needed to add sample profile to the nodes...

Comment (pjbgf) [reply test/e2e/node/security_context.go:195]: Just to clarify the test I mentioned:

```
apiVersion: v1
kind: Pod
metadata:
  name: pod-level-container-perms
  annotations:
    seccomp.security.alpha.kubernetes.io/pod: 'localhost/blockall.json'
    container.seccomp.security.alpha.kubernetes.io/matey-container: 'unconfined'
spec:
  containers:
  - name: matey-container
    image: alpine
    command: ["sh"]
    args: ["-c", "echo hi && sleep 120"]
```
Using profile:
```
{
    "defaultAction": "SCMP_ACT_ERRNO"
}
```

This fails to start - which is the expected behaviour - with the following error:

```
4s          Warning   FailedCreatePodSandBox    pod/pod-level-container-perms   Failed to create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "pod-level-container-perms": Error response from daemon: OCI runtime start failed: cannot start a container that has stopped: unknown
```

If the seccomp is not applied to the sandbox container, the pod would execute normally.

File: test/e2e/node/security_context.go
Comment (pjbgf) [inline test/e2e/node/security_context.go:165]: nit:
```suggestion
	ginkgo.It("should support seccomp runtime/default by annotation [LinuxOnly]", func() {
```

**Review by pjbgf — APPROVED** [review]
Really good stuff, @saschagrunert, thank you for this contribution!

/lgtm

File: test/e2e/node/security_context.go
Comment (saschagrunert) [reply test/e2e/node/security_context.go:165]: :+1: fixed

**Comment by saschagrunert** [comment]
@kubernetes/sig-testing @kubernetes/sig-node-pr-reviews @pjbgf @hasheddan please take a look, I think it would be valuable to get this in for v1.19.0.

File: test/e2e/node/security_context.go
Comment (pjbgf) [inline test/e2e/node/security_context.go:172]: nit: maybe this better translates the intention behind this test
```suggestion
	ginkgo.It("should overwrite pod-level profile with 'unconfined' from container seccomp profile field [LinuxOnly]", func() {
```

File: test/e2e/node/security_context.go
Comment (pjbgf) [inline test/e2e/node/security_context.go:187]: nit: maybe this better translates the intention behind this test
```suggestion
	ginkgo.It("should overwrite pod-level profile with 'runtime/default' from container seccomp profile field [LinuxOnly]", func() {
```

**Review by pjbgf — APPROVED** [review]
@saschagrunert thank you for improving the seccomp e2e test coverage!

I am suggesting two small nits for your consideration, but overall I am happy to approve as it is.

/lgtm

File: test/e2e/node/security_context.go
Comment (saschagrunert) [reply test/e2e/node/security_context.go]: Fixed as suggested :+1:

File: test/e2e/node/security_context.go
Comment (saschagrunert) [reply test/e2e/node/security_context.go]: Fixed as suggested :+1:

**Comment by liggitt** [comment]
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/91442/pull-kubernetes-node-e2e/1282929582286049280 flake is #93210

**Comment by saschagrunert** [comment]
Kind ping for @kubernetes/sig-testing @kubernetes/sig-node-pr-reviews to request review.

**Comment by saschagrunert** [comment]
> /assign

Thank you @dashpole! :pray:

**Comment by saschagrunert** [comment]
@kubernetes/milestone-maintainers do we wanna put those tests into v1.19 with respect to the seccomp GA implementation? From my point of view it would be good to have them in the upcoming release.

**Comment by neolit123** [comment]
i'm surprised the e2e tests for a GA feature did not merge in time.

it would has to be an exception for code-freeze.
having them in 1.19 seems right, but maybe the decision is up to SIG Node and the Release 1.19 CI team if they want this last minute with respect to potential flakes before the release.

**Comment by liggitt** [comment]
Many of these tests are exercising things we already exercise via unit tests. I don't think we need to add an e2e test around every combination of annotation and field and can rely on the fact that the kubelet honors them both (that was largely the point of making the two consistent on pod creation)

**Comment by liggitt** [comment]
Changing the existing tests to use the API field would make more sense to me

**Comment by saschagrunert** [comment]
> Changing the existing tests to use the API field would make more sense to me

Sure, changed the tests to only use the new fields instead of the annotations.

File: test/e2e/node/security_context.go
Comment (liggitt) [inline test/e2e/node/security_context.go:149]: change the descriptions to drop "annotation"

Comment (saschagrunert) [reply test/e2e/node/security_context.go]: Yep fxed, dropped the `alpha` as well. :+1:

**Comment by liggitt** [comment]
this will be moved back into the milestone as part of the [phased reopen for 1.20 merges](https://docs.google.com/document/d/1yI7o7R2bOB7QtkHUFMHMb5XObiJmzIAnJPuxhdAX-CA/edit)

**Pull request merged**
